### ISSUE TITLE
2023-09 UI Freezes

### ISSUE THREAD

**Issue by gigermocas** [eclipse-platform/.github#157]
Since upgrading my Eclipse 2023-06 installation to 2023-09, I'm seeing constant large UI Freezes every time I interact with a Java editor, copying/paste, CTRL commands to drill into functions, etc.
The other plugin I've updated at the same time was Spring's STS4, but I've disabled all language servers/reconcilers and I'm still seeing this.

I've attached the log of one of the  freezes I've experienced (trace was to long to paste here)
[trace.log](https://github.com/eclipse-platform/.github/files/12748766/trace.log)

Please advise if more information is required.

**Comment by vogella** [comment]
AFAICS you have freezes in the update manager and the JDT file access. I think @jukzi and @laeubi did significant changes in the latest SDK in this area. Can you try the latest I-build from https://download.eclipse.org/eclipse/downloads/ (currently [I20230927-1800](https://download.eclipse.org/eclipse/downloads/drops4/I20230927-1800/)). Just download it, use File -> Import -> Install -> From existing installation to install of the plug-ins you need additional and try again?

**Comment by gigermocas** [comment]
I don't see a difference in the behavior with the new plug-ins, I got a 12s freeze by selecting some java code:
[freeze2.log](https://github.com/eclipse-platform/.github/files/12750217/freeze2.log)

**Comment by vogella** [comment]
@jukzi looks to me like a freeze in JDT. If you agree, can you transfer the issue to JDT? Or is it not possible between the Eclipse repos? I can't see if the transfer is possible as I'm not a JDT committer.

**Comment by gigermocas** [comment]
FYI, I'm also seeing a bunch of "Timeout waiting for data to generate LS hover" messages in the log (maybe related):
```
java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at org.eclipse.lsp4e.jdt.LSJavaHoverProvider.getHoverInfo2(LSJavaHoverProvider.java:65)
	at org.eclipse.jdt.internal.ui.text.java.hover.BestMatchHover.getHoverInfo2(BestMatchHover.java:163)
	at org.eclipse.jdt.internal.ui.text.java.hover.BestMatchHover.getHoverInfo2(BestMatchHover.java:130)
	at org.eclipse.jdt.internal.ui.text.java.hover.JavaEditorTextHoverProxy.getHoverInfo2(JavaEditorTextHoverProxy.java:89)
	at org.eclipse.jface.text.TextViewerHoverManager$1.run(TextViewerHoverManager.java:155)
```

**Comment by jukzi** [comment]
> I don't see a difference in the behavior with the new plug-ins, I got a 12s freeze by selecting some java code: [freeze2.log](https://github.com/eclipse-platform/.github/files/12750217/freeze2.log)

```
Stack Trace
	at java.base@17.0.8.1/java.io.WinNTFileSystem.canonicalize0(Native Method)
	at java.base@17.0.8.1/java.io.WinNTFileSystem.canonicalize(WinNTFileSystem.java:462)
	at java.base@17.0.8.1/java.io.File.getCanonicalPath(File.java:626)
	at org.eclipse.aether.internal.impl.DefaultTrackingFileManager.getMutex(DefaultTrackingFileManager.java:127)
	at org.eclipse.aether.internal.impl.DefaultTrackingFileManager.read(DefaultTrackingFileManager.java:57)
	at org.eclipse.aether.internal.impl.EnhancedLocalRepositoryManager.readRepos(EnhancedLocalRepositoryManager.java:220)
	at org.eclipse.aether.internal.impl.EnhancedLocalRepositoryManager.checkFind(EnhancedLocalRepositoryManager.java:151)
	at org.eclipse.aether.internal.impl.EnhancedLocalRepositoryManager.find(EnhancedLocalRepositoryManager.java:128)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:359)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:260)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:242)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:231)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:172)
	at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.resolveCachedArtifactDescriptor(DfDependencyCollector.java:382)
	at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.getArtifactDescriptorResult(DfDependencyCollector.java:368)
	at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.processDependency(DfDependencyCollector.java:218)
	at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.processDependency(DfDependencyCollector.java:156)
	at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.process(DfDependencyCollector.java:138)
	at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.doCollectDependencies(DfDependencyCollector.java:108)
	at org.eclipse.aether.internal.impl.collect.DependencyCollectorDelegate.collectDependencies(DependencyCollectorDelegate.java:222)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:87)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.collectDependencies(DefaultRepositorySystem.java:305)
	at org.apache.maven.project.DefaultProjectDependenciesResolver.resolve(DefaultProjectDependenciesResolver.java:151)
	at org.apache.maven.project.DefaultProjectBuilder.resolveDependencies(DefaultProjectBuilder.java:224)
	at org.apache.maven.project.DefaultProjectBuilder.build(DefaultProjectBuilder.java:640)
	at org.apache.maven.project.DefaultProjectBuilder.build(DefaultProjectBuilder.java:375)
	at org.eclipse.m2e.core.internal.IMavenToolbox.readMavenProjects(IMavenToolbox.java:117)

```

getCanonicalPath is very slow on new JDK. Please report to org.eclipse.aether and ask them to not use getCanonicalPath. Typically getCanonicalPath() it's not what you need.
See for example https://bugs.eclipse.org/bugs/show_bug.cgi?id=571614

Closing as not-platform.

**Comment by gigermocas** [comment]
Maybe we're seeing different problems here, I guess the fault is mine for the confusion. The vast majority of freezes I'm seeing are on the jdt, the first trace I posted here doesn't have references to either aether (something to do with artifact resolution!?) or getCanonicalPath(). I didn't check if the freeze in freeze2.log was for the same specific problem I was experiencing with a fresh 2023-09... all I was seeing were the ui freezes, apologies for that.

I installed a fresh 2023-09 and updated Eclipse Java Development Tools to 3.19.300.v20231108-0715, captured traces similar to the first one:
[trace4.log](https://github.com/eclipse-platform/.github/files/13307599/trace4.log)
[trace5.log](https://github.com/eclipse-platform/.github/files/13307795/trace5.log)


I'm developing with 2023-06, as 2023-09 is currently unusable for me,
